src/game/traps/lava.c

```diff
@@ -75,9 +75,24 @@ void LavaBurn(ITEM_INFO *item)
     } while (!(type & END_BIT));
 
     if (!on_lava) {
+        return false;
+    } else {
+        return true;
+    }
+}
```

Comment:
Please make this function exit early. In other words refactor it to `return true` as soon as possible within its main loop, and `return false` at the end of the function as a fallback value.

Off-topic for educational purposes (doesn't apply after the suggestion above):

```c
if (!on_lava) {
    return false;
} else {
    return true;
}
```

This snippet is quite low quality. First, since this snippet has both `if` and `else`, and both of the branches are one-liners, there's no need for the negation - it only reduces readability. So let's flip it:

```c
if (on_lava) {
    return true;
} else {
    return false;
}
```

This is still wordy. Both of the branches evaluate to the same statement – `return x`, so we can turn it into a ternary expression. We get:

```c
return on_lava ? true : false;
```

But hopefully you can see that this is silly and can be just represented as:

```c
return on_lava;
```

However, like I said earlier, there's no need to run the entire loop – we can know the state of `on_lava` much earlier. And making the function exit as early as possible is what we call early exits. It's not an optimization technique, it's to boost readability - the `return` statements get highlighted in any decent IDE and you can easily see the control flow.


Don't forget to remove the no longer necessary `on_lava` variable after the fact :)